iroh-net/src/bin/derper.rs

```diff
@@ -284,10 +284,24 @@ impl Config {
         if !path.as_ref().is_file() {
             bail!("config-path must be a valid toml file");
         }
-        let config_ser = tokio::fs::read_to_string(path)
+        let mut config_ser = tokio::fs::read_to_string(&path)
             .await
             .context("unable to read config")?;
-        let config = toml::from_str(&config_ser).context("unable to decode config")?;
+        let mut secret_key_injected = false;
+        if !config_ser.contains("secret_key") {
```

Comment:
I would strongly prefer to inject this after toml parsing, there is actually a nice way to do this. 

```rust
#[serde(default = "SecretKey::generate")]
secret_key: SecretKey,
```

will do exactly that.

Ref https://serde.rs/field-attrs.html#default

Reply:
I swear serde is more powerful than any AI we have so far :D